app/screens/ProductList/ProductList.js

```diff
@@ -38,6 +44,13 @@ class ProductList extends Component {
             });
     }
 
+    onScrollHandlerAnimation = () => {
```

Comment:
To make this animation to work put Animated.event directly to onScroll:
```
...
onScroll={Animated.event(
            [{ nativeEvent: { contentOffset: { y: this.flatListValue }} }],
            { useNativeDriver: true }
        )}
...
```
This happens due to binding your flatListValue to the native code. 
Or instead you can use method with is return Animated.event but without `useNativeDriver: true`

```
        onScrollHandlerAnimation = () => Animated.event(
            [{ nativeEvent: { contentOffset: { y: this.flatListValue }} }],
            { useNativeDriver: true }
        );
    
 onScroll={this.onScrollHandlerAnimation}
```

p.s. first option is preferable